### ISSUE TITLE
Error: Unexpected discovery message

### ISSUE THREAD

**Issue by kopiro** [dhleong/playactor#46]
Hey; I've implemented your library in https://github.com/kopiro/homebridge-playstation,  and sometimes I'm getting:

```
Error: Unexpected discovery message:
    at Object.parseMessage (/usr/lib/node_modules/homebridge-playstation/node_modules/playactor/dist/discovery/messages.js:30:19)
    at Socket.<anonymous> (/usr/lib/node_modules/homebridge-playstation/node_modules/playactor/dist/discovery/udp.js:114:39)
    at Socket.emit (node:events:402:35)
    at UDP.onMessage (node:dgram:939:8)
```

This happens after `connection.close()` has been called and we call `Device::withId().discover()` again;

The weird thing is that this exception goes through the entire main thread and causes homebridge to crash

**Comment by dhleong** [comment]
Interesting. Not sure what's sending that invalid message, but we should definitely not crash. Should be easy enough to fix, just need to log and skip un-parseable messages in `UdpDiscoveryNetworkFactory`....

**Comment by kopiro** [comment]
I may think this problem could happen whenever you do `Device::withId()` and immediately after you call `.openConnection()` WITHOUT calling `.discover()`

**Comment by kopiro** [comment]
@dhleong https://github.com/dhleong/playactor/pull/49

**Comment by dhleong** [comment]
Fixed by #49 thanks!
